frontend/utilities/sql_tools.ts

```diff
@@ -99,7 +99,7 @@ export const parseSqlTables = (
       ) {
         const targetName = ((node as unknown) as ISqlCteNode).target.name;
         cteTables.push(targetName);
-      } else if (node.variant === "table") {
+      } else if (node.variant === "table" && !(node.type === "function")) {
```

Comment:
wdyt about refactoring the callback function into something like this to reduce potential gotchas from the type coercions? 

```ts
  const _callback = (node: IAstNode) => {
    if (!node) {
      return;
    }

    if (
      (node.variant === "common" || node.variant === "recursive") &&
      node.format === "table" &&
      node.type === "expression"
    ) {
      const targetName = node.target && (node.target as IAstNode).name;
      targetName &&
        typeof targetName === "string" &&
        cteTables.push(targetName);
      return;
    }

    node.variant === "table" &&
      // ignore table-valued functions (see, e.g., https://www.sqlite.org/json1.html#jeach)
      node.type !== "function" &&
      node.name &&
      typeof node.name === "string" &&
      results.push(node.name);
  };
```

Reply:
I like where you're going. Are you sure you want to abandon the more specific node types?

Reply:
In hindsight, I think they were a naive approach on my part and gave a false sense of security. We could rewrite them in a stricter sense, but since they were only being used in these two places, I wasn't sure it was worth the trouble.